File: OOP Using Java/Standard Searcing And Sorting/QuickSort.java
public class QuickSort{
    void sort(String arr[], int low, int high){
        if (low < high) {
            int pi = partition(arr, low, high);
            sort(arr, low, pi - 1);
            sort(arr, pi + 1, high);
        }
    }

    int partition(String arr[], int low, int high){
        String pivot = arr[low];
        int i = high + 1;

        for (int j = high; j > low; j--) {
            if (arr[j].compareToIgnoreCase(pivot) >= 0){
                i--;
                swap(arr, i, j);
            }
        }
        swap(arr, i - 1, low);
        return i - 1;
    }

    void swap(String arr[], int i, int j){
        String temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    static void printArray(String arr[]){
        for (String str : arr)
            System.out.print(str + " ");
        System.out.println();
    }

    public static void main(String args[]){
        String arr[] = {"John", "Alice", "Bob", "David", "Eva"};
        int n = arr.length;

        System.out.println("Before sorting:");
        for(int i=0;i<n;i++)
            System.out.print(arr[i]+" ");
        System.out.println();

        QuickSort ob = new QuickSort();
        ob.sort(arr, 0, n - 1);

        System.out.println("Sorted array:");
        printArray(arr);
    }
}

File: Data Structures/Cycle 4/Sorting Algorithms/quickSort.c
#include <stdio.h>

int arr[50], n , search;

void display(int ar[], char* type) {
    printf("%s: {", type);
    for (int i = 0; i < n; i++)
        printf("\t%d", ar[i]);
    printf("\t}\n");
}

void initArr(){
    int temp;
    printf("Enter The Number Of Elements: ");
    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        printf("Enter The Element %d: ", i + 1);
        scanf("%d", &arr[i]);
    }
    display(arr, "Original Array");
}

int partition(int arr[100], int low, int high) {
    int pivot = arr[low];
    int i = low, j = high;
    while (i < j) {
        while (pivot >= arr[i])
            i++;
        while (pivot < arr[j])
            j--;
        if (i < j) {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
    arr[low] = arr[j];
    arr[j] = pivot;
    return j;
}

void quickSort(int arr[100], int low, int high) {
    if (low < high) {
        int pivot = partition(arr, low, high);
        quickSort(arr, low, pivot - 1);
        quickSort(arr, pivot + 1, high);
    }
}

void main(){
    initArr();
    quickSort(arr, 0, n - 1);
    display(arr , "Quick Sort");
}